wp/headless-wp/includes/classes/Integrations/PolylangHrefLangs.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * HeadlessWP PolylangHrefLangs Class.
+ * Adds hreflangs to Yoast head
+ *
+ * @package HeadlessWP
+ */
+
+namespace HeadlessWP\Integrations;
+
+use Yoast\WP\SEO\Presenters\Abstract_Indexable_Presenter;
+use HeadlessWP\Utils\Rest;
+use HeadlessWP\Utils\Langs;
+
+/**
+ * The PolylangHrefLangs integration class
+ */
+class PolylangHrefLangs extends Abstract_Indexable_Presenter {
+
+	/**
+	 * Output the hreflang tags if they exist.
+	 */
+	public function present() {
+		if ( function_exists( 'pll_languages_list' ) ) {
+			$hreflangs = $this->get();
+
+			if ( ! empty( $hreflangs ) ) {
+				return $hreflangs;
+			}
+		}
+
+		return '';
+	}
+
+	/**
+	 * Get hreflang tags for the current post.
+	 * https://polylang.pro/hreflang-tag-attributes-and-polylang-everything-you-need-to-know
+	 *
+	 * @return string Hreflang tags or empty string.
+	 */
+	public function get() {
+
+		$languages = pll_languages_list();
+
+		if ( 2 > count( $languages ) ) {
+			return '';
+		}
+
+		$hreflangs_output = '';
+		$hreflangs_urls   = array();
+
+		$post_id      = get_the_ID();
+		$translations = array();
+		$is_taxonomy  = false;
+		$is_homepage  = false;
+		$term         = null;
+
+		$paged = filter_input( INPUT_GET, 'page', FILTER_SANITIZE_NUMBER_INT );
+
+		// Don't output anything on paged archives: see https://wordpress.org/support/topic/hreflang-on-page2
+		// Don't output anything on paged pages and paged posts
+		if ( 1 < $paged ) {
+			return '';
+		}
+
+		// Get Term or Post translations.
+		if ( empty( $post_id ) ) {
+			$term = Rest\is_taxonomy_rest_request();
+			if ( $term instanceof \WP_term && pll_is_translated_taxonomy( $term->taxonomy ) ) {
+				$translations = pll_get_term_translations( $term->term_id );
+				$is_taxonomy  = true;
+			}
+		} else {
+			$translations = pll_get_post_translations( $post_id );
+		}
+
+		if ( empty( $translations ) ) {
+			return '';
+		}
+
+		$default_lang = Langs\get_default_locale_slug();
+
+		// Herflangs for archive first pages.
+		if ( $is_taxonomy ) {
+			foreach ( $translations as $language => $translated_id ) {
+				// Get term object for translations and only include hreflang tags if term has posts.
+				$term_obj = $translated_id === $term->term_id ? $term : get_term( $translated_id, $term->taxonomy );
+				if ( 0 === $term_obj->count ) {
+					continue;
+				}
+
+				$hreflangs_urls[ $language ] = get_term_link( $translated_id );
+			}
+		} else {
+			$homepages = $this->get_homepage_post_ids();
```

Comment:
actually this is not needed, missed that Polylang has a setting for this, I'll update

Reply:
fixed the overriding polylang setting for homepage slug